apis/Google.Cloud.PubSub.V1/Google.Cloud.PubSub.V1/SubscriberClient.cs

```diff
@@ -530,297 +533,605 @@ public Flow(long maxByteCount, long maxElementCount, Action<Task> registerTaskFn
                     {
                         _event.Set();
                     }
-                })));
+                }));
+                _registerTaskFn(task);
+            }
+        }
+
+        // internal for testing.
+        internal class RequeueableQueue<T>
```

Comment:
A doc comment would be helpful :) So it looks like there's the queued data, and the requeued data. If you requeue more data, it comes after all previously-requeued data, but before queued data. Is that right?